pkg/kubectl/service_basic.go

```diff
@@ -225,12 +226,25 @@ func (s ServiceCommonGeneratorV1) StructuredGenerate() (runtime.Object, error) {
 		})
 	}
 
-	// setup default label and selector
+	// setup default label
 	labels := map[string]string{}
 	labels["app"] = s.Name
-	selector := map[string]string{}
-	selector["app"] = s.Name
-
+	// setup selector according to selector flag, use default if nothing is provided
+	// selector will not be added for service type ExternalName
+	isTypeExternalName := s.Type == api.ServiceTypeExternalName
+	selectorString := s.Selector
+	var selector map[string]string
+	if !isTypeExternalName {
+		if len(selectorString) == 0 {
```

Comment:
@shiywang : That may be possible, based on @smarterclayton description, I thought default selector "app=name" is not should not be present if a custom selector is passed to the command. I thing merging the custom selector with default will cause confusion
--selector='a=b,c=d'  ------will result in ----> 'app=name,a=b,c=d'
@smarterclayton can you please suggest